curriculum/challenges/english/25-front-end-development/lab-react-forms/67d936de7055982b02baf186.md

```diff
@@ -0,0 +1,739 @@
+   const stateWithFalse = getInnerTextExcept("input,script");
+
+   inp = document.querySelector(`input[type="checkbox"]`);
+   await React.act(async () => {
+     const setter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "checked").set;
+     setter.call(inp, true);
+     inp.dispatchEvent(new Event("click", { bubbles: true}));
+
+     const submitBtn = document.querySelector("form button, [type='submit']");
+     submitBtn.click();
+   });
+
+   const stateWithTrue = getInnerTextExcept("input,script");
+
+   assert.notEqual(stateWithFalse, stateWithTrue);
+}
+```
+
+# --before-all--
+
+```js
+async function adjustAllRequired() {
+  const inps = document.querySelectorAll("input[required], select[required], textarea[required]");
+  inps.forEach((inp) => {
+    let setter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
+    switch (inp.type) {
+      case "text":
+        setter.call(inp, "required text");
+        inp.dispatchEvent(new Event("change", { bubbles: true}));
+        break;
+      case "email":
+        setter.call(inp, "required-email@freecodecamp.org");
+        inp.dispatchEvent(new Event("change", { bubbles: true}));
+        break
+      case "number":
+        setter.call(inp, 1);
+        inp.dispatchEvent(new Event("change", { bubbles: true}));
+        break;
+      case "checkbox":
+        setter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "checked").set;
+        setter.call(inp, true);
+       inp.dispatchEvent(new Event("click", { bubbles: true}));
+        break;
+    }
+    inp.dispatchEvent(new Event("change", { bubbles: true}));
+  });
+}
+
+function getInnerTextExcept(removingSelector) {
+  const body = document.body.cloneNode(true);
+
+  const squareElements = body.querySelectorAll(removingSelector);
+  squareElements.forEach(element => {
+    element.parentNode.removeChild(element)
+  });
+
+  return body.innerText;
+}
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="UTF-8" />
+    <title>Event RSVP</title>
+    <link rel="stylesheet" href="styles.css" />
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/18.3.1/umd/react.development.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/18.3.1/umd/react-dom.development.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/7.26.5/babel.min.js"></script>
+</head>
+
+<body>
+    <div id="root"></div>
+    <script
+      data-plugins="transform-modules-umd"
+      type="text/babel"
+      data-presets="react"
+      data-type="module"
+    >
+      import { EventRSVPForm } from './index.jsx';
+      ReactDOM.createRoot(document.getElementById('root')).render(<EventRSVPForm />);
+    </script>
+</body>
+
+</html>
+```
+
+```css
+
+```
+
+```jsx
+export function EventRSVPForm() {
+
```

Comment:
we need to provide `const { useState } = React;` for them. 

Otherwise campers will try to do `import {useState} from "react"` and be surprised to see this in the console when they try to debug their code. 
<img width="954" alt="Screenshot 2025-04-07 at 9 02 56 AM" src="https://github.com/user-attachments/assets/3dffb30b-47ec-4d31-8a43-c5bf6696e0ef" />